src/main/java/codeu/model/store/persistence/PersistentDataStore.java

```diff
@@ -188,6 +191,51 @@ public Conversation loadActFeedConversation() throws PersistentDataStoreExceptio
     return messages;
   }
 
+	  /**
+   * Loads all Activity objects from the Datastore service and returns them in a List, sorted in
+   * ascending order by creation time.
+   *
+   * @throws PersistentDataStoreException if an error was detected during the load from the
+   *     Datastore service
+   */
+  public List<Activity> loadActivities() throws PersistentDataStoreException {
+
+    List<Activity> activities = new ArrayList<>();
+
+    // Retrieve all activities from the datastore.
+    Query query = new Query("activity-objects").addSort("creation_time", SortDirection.ASCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      try {
+				UUID activityUuid = UUID.fromString((String) entity.getProperty("activity_uuid"));
+        int allTimeCount = (int) Integer.parseInt((String) entity.getProperty("allTimeCount"));
+        Instant creationTime = Instant.parse((String) entity.getProperty("creation_time"));
+				String message = (String) entity.getProperty("message");
+				UUID userUuid = UUID.fromString((String) entity.getProperty("user_uuid"));
+        String username = (String) entity.getProperty("username");
+        Type type = Type.valueOf((String) entity.getProperty("type"));
+        UUID conversationId;
+        if (entity.getProperty("conversation_uuid") == null) {
```

Comment:
Better (less verbose) would be to initialize conversationId to null then reassign it if the property isn't null. Declaring a local variable without initializing it is rarely a good thing to do.